pkg/stub/handler.go

```diff
@@ -282,12 +377,17 @@ func (h *Handler) CreateDefaultResourceIfNeeded() error {
 			}
 			logrus.Println("got already exists error on create default, just going to forgo this time vs. retry")
 			// just return the raw error and initiate the sdk's requeue/ratelimiter setup
-			return err
 		}
-		return h.GoodConditionUpdate(srcfg, corev1.ConditionTrue, v1alpha1.SamplesExist)
+		h.deleteInProgress = false
+		return err
+	} else {
+		if deleteInProgress {
+			return h.processError(srcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "issues waiting for delete to complete: %v")
+		} else {
+			logrus.Println("SampleResource already received, not creating default")
```

Comment:
and you should do this return on line 351 (if srccfg!=nil)

Reply:
```
if deleteinprogress {
  all logic related to waiting for delete, returning if we get an error waiting for deletion
  set h.SamplesResource=nil  // any reason not to do this here?
}
if h.samplesResource!=nil {
  return nil // we already have a samples resource
}

// logic to create default sampleresource.
```

Reply:
moved setting of h.SamplesResource to nil at that spot, worked fine, along with the rest of this